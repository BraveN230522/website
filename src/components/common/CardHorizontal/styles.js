import styled from '@emotion/styled';

const CardHorizontalStyles = styled.div`
  a {
    background-clip: border-box;
    background-color: ${({ theme }) => theme.colors.whiteSmoke};
    border-radius: 8px;
    display: flex;
    flex-direction: row;
    position: relative;
    width: 447px;
    text-decoration: none;
    &:hover {
      text-decoration: none;
      cursor: pointer;
    }
  }
`;

const CardHorizontalImg = styled.img`
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  object-fit: cover;
  max-width: 200px;
  min-width: 200px;
  max-height: 167px;
`;

const CardBody = styled.div`
  padding: 15px 20px;
  min-height: 166px;

  ${({ theme }) => theme.media.phone`
    padding: 10px 15px;
  `}
`;

const CardHorizontalTitle = styled.h1`
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.fonts.Inter};
  font-size: 17px;
  font-weight: 700;
  line-height: 23px;
  margin-top: 0;
  margin-bottom: 12px;
  letter-spacing: -0.1px;
  text-overflow: ellipsis;
  /* overflow: hidden;
   display: -webkit-box;
   -webkit-line-clamp: 2;
   -webkit-box-orient: vertical; */
`;

const CardHorizontalInfo = styled.p`
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.fonts.Helvetica};
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
  letter-spacing: -0.08px;
  margin-bottom: 0;
  height: 65px;
  width: 208px;
  overflow: hidden;
  text-overflow: ellipsis;
  ${({ theme }) => theme.media.phone`
    max-width: 140px;
  `}
`;

export { CardHorizontalStyles, CardHorizontalImg, CardBody, CardHorizontalTitle, CardHorizontalInfo };
