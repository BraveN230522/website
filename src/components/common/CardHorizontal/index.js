import React from 'react';

import { CardHorizontalStyles, CardHorizontalImg, CardBody, CardHorizontalTitle, CardHorizontalInfo } from './styles';

const CardHorizontal = ({ cardData }) => {
  return (
    <CardHorizontalStyles>
      <a href={cardData.link} target="_blank" rel="noopener noreferrer">
        <CardHorizontalImg src={cardData.image} alt={cardData.title} />
        <CardBody>
          <CardHorizontalTitle>{cardData.title}</CardHorizontalTitle>
          {cardData.info && <CardHorizontalInfo>{cardData.info}</CardHorizontalInfo>}
        </CardBody>
      </a>
    </CardHorizontalStyles>
  );
};

export default CardHorizontal;
