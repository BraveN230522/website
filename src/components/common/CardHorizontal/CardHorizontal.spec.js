import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import CardHorizontal from './index';

const mockCardHorizontal = [
  {
    id: 1,
    img:
      'https://travel.home.sndimg.com/content/dam/images/travel/fullrights/2017/8/0/CI_Juniors-49thBroadwayInterior.jpg.rend.hgtvcom.616.462.suffix/1501638474622.jpeg',
    title: 'Restaurants near Madison Square Garden',
    info:
      'For years the area around Manhattan’s premier entertainment venue has been criticised for a lack of decent food options.',
  },
  {
    id: 2,
    img:
      'https://content.r9cdn.net/rimg/kimg/f1/0b/34800236-58e51e7e.jpg?width=1004&crop=true&caller=HotelDetailsPhoto',
    title: 'Hotels near Madison Square Garden',
    info:
      'Now, you might have heard that Manhattan isn’t exactly the cheapest place in the world, and a night or two in a hotel could ..',
  },
];

describe('<CardHorizontal />', () => {
  it('Renders a card horizontal component', () => {
    const { container } = render(
      mockCardHorizontal.map(item => <CardHorizontal key={item.id} cardData={item} />),
      {
        wrapper: withGlobals,
      }
    );
    expect(container).toMatchSnapshot();
  });
});
