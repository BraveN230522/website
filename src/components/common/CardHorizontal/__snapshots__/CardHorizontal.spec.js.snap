// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardHorizontal /> Renders a card horizontal component 1`] = `
.emotion-8 a {
  background-clip: border-box;
  background-color: #f4f4f4;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  width: 447px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-8 a:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-0 {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  object-fit: cover;
  max-width: 200px;
  min-width: 200px;
  max-height: 167px;
}

.emotion-6 {
  padding: 15px 20px;
}

@media (max-width:48em) {
  .emotion-6 {
    padding: 10px 15px;
    max-width: 167px;
  }
}

.emotion-2 {
  color: #000000;
  font-family: Inter,sans-serif;
  font-size: 17px;
  font-weight: 700;
  line-height: 23px;
  margin-top: 0;
  margin-bottom: 12px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  text-overflow: ellipsis;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

.emotion-4 {
  color: #000000;
  font-family: Helvetica,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
  -webkit-letter-spacing: -0.08px;
  -moz-letter-spacing: -0.08px;
  -ms-letter-spacing: -0.08px;
  letter-spacing: -0.08px;
  margin-bottom: 0;
  height: 65px;
  width: 208px;
  overflow: hidden;
  text-overflow: ellipsis;
}

@media (max-width:48em) {
  .emotion-4 {
    max-width: 140px;
  }
}

<div>
  <div
    class="emotion-8 emotion-9"
  >
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        class="emotion-0 emotion-1"
      />
      <div
        class="emotion-6 emotion-7"
      >
        <h1
          class="emotion-2 emotion-3"
        >
          Restaurants near Madison Square Garden
        </h1>
        <p
          class="emotion-4 emotion-5"
        >
          For years the area around Manhattan’s premier entertainment venue has been criticised for a lack of decent food options.
        </p>
      </div>
    </a>
  </div>
  <div
    class="emotion-8 emotion-9"
  >
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        class="emotion-0 emotion-1"
      />
      <div
        class="emotion-6 emotion-7"
      >
        <h1
          class="emotion-2 emotion-3"
        >
          Hotels near Madison Square Garden
        </h1>
        <p
          class="emotion-4 emotion-5"
        >
          Now, you might have heard that Manhattan isn’t exactly the cheapest place in the world, and a night or two in a hotel could ..
        </p>
      </div>
    </a>
  </div>
</div>
`;
